### PR TITLE
feat: Add variables on Badge Title and Description

### PR DESCRIPTION
**Pull request by FedeMadoery**
# Description

Allow TP Creators to configure their badge titles and descriptions, with their own text and also add the possibility to each minter be able to put his own data on it with dynamic variables like: {{studentName}} or {{displayName}}

### CODE REVIEW

File: src/hooks/utils/useListHighlightItem.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          }
+
+          const itemRef = listRefs[index].current
+          if (itemRef) {
+            itemRef.scrollIntoView({
+              behavior: 'smooth',
+              block: 'end',
+              inline: 'center',
+            })
+          }
```
Comment (ramabit) [inline src/hooks/utils/useListHighlightItem.tsx:51]: this duplicated code can be refactored in a function

File: src/pagePartials/badge/mint/steps/dynamicForm/thirdParty/DynamicRequiredData.tsx [hunk 2]
```diff
@@ -33,9 +39,21 @@ export default function DynamicRequiredData({
   )
 
   const handleSubmitNext = (data: z.infer<typeof RequiredDataSchema>) => {
+    const autoFilledData = generateAutofilledValues(
+      {
+        [ReplacementKeys.issueDate]: dayjs().format('DD/MM/YYYY'),
+        [ReplacementKeys.expirationDate]: expirationTime
+          ? dayjs().add(expirationTime, 'seconds').format('DD/MM/YYYY')
+          : 'Never expires',
```
Comment (ramabit) [inline src/pagePartials/badge/mint/steps/dynamicForm/thirdParty/DynamicRequiredData.tsx:47]: need an i18n key here?

File: src/pagePartials/badge/mint/steps/dynamicForm/thirdParty/DynamicRequiredData.tsx [hunk 3]
```diff
@@ -33,9 +39,21 @@ export default function DynamicRequiredData({
   )
 
   const handleSubmitNext = (data: z.infer<typeof RequiredDataSchema>) => {
+    const autoFilledData = generateAutofilledValues(
+      {
+        [ReplacementKeys.issueDate]: dayjs().format('DD/MM/YYYY'),
```
Comment (ramabit) [inline src/pagePartials/badge/mint/steps/dynamicForm/thirdParty/DynamicRequiredData.tsx:44]: should we have a central place (perhaps env variable or util) where we setup the date format for all things and can be easily changed later if needed?

File: src/hooks/subgraph/useBadgeModelThirdPartyMetadata.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import useSWR from 'swr'
 import { BadgeModelHooksOptions } from '@/src/hooks/subgraph/types'
 import useSubgraph from '@/src/hooks/subgraph/useSubgraph'
 import { getFromIPFS } from '@/src/hooks/subgraph/utils'
-import { useWeb3Connection } from '@/src/providers/web3ConnectionProvider'
+const { useWeb3Connection } = await import('@/src/providers/web3ConnectionProvider')
```
[the remark this answers is not in the document]
Comment (FedeMadoery) [reply src/hooks/subgraph/useBadgeModelThirdPartyMetadata.tsx:6]: Bc in this way is bundled apart of the page, and all the places are going to consume the same chunk for this hook, if not some of the use the chunk and some other bundled it inside of the page

File: src/pagePartials/badge/mint/steps/dynamicForm/thirdParty/DynamicRequiredData.tsx (hunk 3 above)
Comment (FedeMadoery) [reply src/pagePartials/badge/mint/steps/dynamicForm/thirdParty/DynamicRequiredData.tsx:44]: Yeah good call, we can do it in another tickets to keep those changes in just one PR

**Review by ramabit — APPROVED** [review]

**Pull request merged**